src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryImpl.java

```diff
@@ -144,19 +145,32 @@ public void saveArtifactFile(CacheResult cacheResult, org.apache.maven.artifact.
      * @return null or content
      */
     @Nonnull
-    public Optional<byte[]> getResourceContent(String url) throws IOException {
+    public Optional<byte[]> getResourceContent(String url) {
+        String fullUrl = getFullUrl(url);
         try {
-            LOGGER.info("Downloading {}", getFullUrl(url));
+            LOGGER.info("Downloading {}", fullUrl);
             GetTask task = new GetTask(new URI(url));
             transporter.get(task);
             return Optional.of(task.getDataBytes());
-        } catch (Exception e) {
-            LOGGER.info("Cannot download {}", getFullUrl(url), e);
+        } catch (ResourceDoesNotExistException e) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Cache item not found: {}", fullUrl, e);
```

Comment:
I think it should be `LOGGER.info`

Reply:
@gnodet there is `if (LOGGER.isDebugEnabled())` and else `info` and, more important without the exception as the last parameter, which is very verbose. per default, we will be fine. 
But could be really more simple to remove this if and using only ` LOGGER.info("Cache item not found: {}", fullUrl);`

Reply:
Maybe? It's a tradeoff - you want to know how well the cache is working but you don't want to clutter the log (in some of our builds we have nearly 100 modules). Also the "not found" phrase implies to me that something could be wrong but of course a cache miss is completely normal, so perhaps "Item not in cache" would sound more innocent. 
A summary at the end of how much the cache was used instead would be very nice...

Reply:
@olamy I'm fine with displaying the exception if debug is enabled, but the level of the message is part of the semantic and should not really change depending on which level is activated.
I've used this pattern a log, but I think it should be:
```
if (isDebugEnabled()) {
   log.info(xxx, exception);
} else {
   log.info(xxx);
}
```

Reply:
@gnodet, why not I'm fine with that. it just looks weird to me :)

Reply:
> Maybe? It's a tradeoff - you want to know how well the cache is working but you don't want to clutter the log (in some of our builds we have nearly 100 modules). Also the "not found" phrase implies to me that something could be wrong but of course a cache miss is completely normal, so perhaps "Item not in cache" would sound more innocent. A summary at the end of how much the cache was used instead would be very nice...

agree I'm using the cache for a build with 328 modules so the current log is very very verbose when the cache not found :)

Reply:
Then the individual statements could be at debug level (with exception at trace level eventually), while the summary could be logged at info level.

Reply:
go with that. But as we are in the case of `ResourceDoesNotExistException` not sure why do we need to have a `if` and display the exception. 
no if but just:
```
} catch (ResourceDoesNotExistException e) {
  LOGGER.info("Cache item not found: {}", fullUrl);
}
```

We are just in the case of a resource not found so no need to print the exception the cause is really clear. I'm not sure to understand of printing a stack even in debug mode. Is there anything more interested in the stack?

Frankly I don't even understand why an exception is used for that. There should be simply a boolean `not found` and not wasting CPU/memory by using an exception (but I know it's a different problem :))

Reply:
could we make progress here? It would be fine to have this done and released as it is very verbose for no gain

Reply:
@AlexanderAshitkin ping :)